### ISSUE TITLE
No such file or directory @ rb_sysopen - install (Errno::ENOENT) during deployment with ruby 3.2.4

### ISSUE THREAD

**Issue by RailsCod3rFuture** [rvm/rvm#5532]
## Description

We've been deploying rails 7/ruby 3.2.4 apps successfully over the last month. Just two days ago, our deployment system failed due to an unknown error that began to occur. We can't seem to spot why RVM started behaving differently for ruby 3.2.4 when we had it running perfectly. Its claiming that the target folder for ruby 3.2.4 doesn't exist when the ruby_executable_hooks file is run. We are using nginx/passenger/puppet/ansible to deploy our apps to rhel8 servers. Not sure if rvm was updated recently. But we are using the latest build - 1.29.12.

## Steps to reproduce
I don't have access to the deployment script. But it seems to fail when groups are assigned to the build user.

## Expected behavior

deployment should complete without failure to passenger.

## Actual behavior

TASK [webservers : Bundle Install on target machines] **************************

`fatal: [appserver.org]: FAILED! => {"changed": true, "cmd": "HTTP_PROXY=http://myproxy.org:8080  rvm use 3.2.4@web-app-gemset --create do bundle install --deployment --without=\"development test jenkins-j6\"", "delta": "0:00:01.669513", "end": "2024-12-17 22:50:38.705950", "msg": "non-zero return code", "rc": 1, "start": "2024-12-17 22:50:37.036437", "stderr": "Warning! PATH is not properly set up, /usr/local/rvm/gems/ruby-3.2.4/bin is not at first place.\n         Usually this is caused by shell initialization files. Search for PATH=... entries.\n         You can also re-add RVM to your profile by running: rvm get stable --auto-dotfiles\n         To fix it temporarily in this shell session run: rvm use ruby-3.2.4\n         To ignore this error add rvm_silence_path_mismatch_check_flag=1 to your ~/.rvmrc file.\n/usr/local/rvm/gems/ruby-3.2.4@web-app-gemset/bin/ruby_executable_hooks:15:in `read': No such file or directory @ rb_sysopen - install (Errno::ENOENT)\n\tfrom /usr/local/rvm/gems/ruby-3.2.4@web-app-gemset/bin/ruby_executable_hooks:15:in `'", "stderr_lines": ["Warning! PATH is not properly set up, /usr/local/rvm/gems/ruby-3.2.4/bin is not at first place.", "         Usually this is caused by shell initialization files. Search for PATH=... entries.", "         You can also re-add RVM to your profile by running: rvm get stable --auto-dotfiles", "         To fix it temporarily in this shell session run: rvm use ruby-3.2.4", "         To ignore this error add rvm_silence_path_mismatch_check_flag=1 to your ~/.rvmrc file.", "/usr/local/rvm/gems/ruby-3.2.4@web-app-gemset/bin/ruby_executable_hooks:15:in `read': No such file or directory @ rb_sysopen - install (Errno::ENOENT)", "\tfrom /usr/local/rvm/gems/ruby-3.2.4@web-app-gemset/bin/ruby_executable_hooks:15:in `'"], "stdout": "", "stdout_lines": []}`

## Environment info
```
[myuser@appserver ~]$ rvm info
Warning! PATH is not properly set up, /usr/local/rvm/gems/ruby-3.2.4/bin is not at first place.
         Usually this is caused by shell initialization files. Search for PATH=... entries.
         You can also re-add RVM to your profile by running: rvm get stable --auto-dotfiles
         To fix it temporarily in this shell session run: rvm use ruby-3.2.4
         To ignore this error add rvm_silence_path_mismatch_check_flag=1 to your ~/.rvmrc file.

ruby-3.2.4:

  system:
    uname:        "Linux myapp.org 4.18.0-553.22.1.el8_10.x86_64 #1 SMP Wed Sep 11 18:02:00 EDT 2024 x86_64 x86_64 x86_64 GNU/Linux"
    name:         "CentOS"
    version:      "8"
    architecture: "x86_64"
    bash:         "/usr/bin/bash => GNU bash, version 4.4.20(1)-release (x86_64-redhat-linux-gnu)"
    zsh:          " => not installed"
    remote_path:  "centos/8/x86_64"

  rvm:
    version:      "1.29.12 (latest)"
    updated:      "1 day 14 hours 23 minutes 14 seconds ago"
    path:         "/usr/local/rvm"
    autolibs:     "[4] Allow RVM to use package manager if found, install missing dependencies, install package manager (only OS X)."

  ruby:
    interpreter:  "ruby"
    version:      "3.2.4"
    date:         "2024-04-23"
    platform:     "x86_64-linux"
    patchlevel:   "2024-04-23 revision af471c0e01"
    full_version: "ruby 3.2.4 (2024-04-23 revision af471c0e01) [x86_64-linux]"

  homes:
    gem:          "/usr/local/rvm/gems/ruby-3.2.4"
    ruby:         "/usr/local/rvm/rubies/ruby-3.2.4"

  binaries:
    ruby:         "/usr/local/rvm/rubies/ruby-3.2.4/bin/ruby"
    irb:          "/usr/local/rvm/rubies/ruby-3.2.4/bin/irb"
    gem:          "/usr/local/rvm/rubies/ruby-3.2.4/bin/gem"
    rake:         "/usr/local/rvm/rubies/ruby-3.2.4/bin/rake"

  environment:
    PATH:         "/home/myuser/.local/bin:/home/myuser/bin:/usr/local/rvm/gems/ruby-3.2.4/bin:/usr/local/rvm/gems/ruby-3.2.4@global/bin:/usr/local/rvm/rubies/ruby-3.2.4/bin:/usr/local/bin:/usr/bin:/usr/local/sbin:/usr/sbin:/opt/puppetlabs/bin:/usr/local/rvm/bin"
    GEM_HOME:     "/usr/local/rvm/gems/ruby-3.2.4"
    GEM_PATH:     "/usr/local/rvm/gems/ruby-3.2.4:/usr/local/rvm/gems/ruby-3.2.4@global"
    MY_RUBY_HOME: "/usr/local/rvm/rubies/ruby-3.2.4"
    IRBRC:        "/usr/local/rvm/rubies/ruby-3.2.4/.irbrc"
    RUBYOPT:      ""
    gemset:       ""
```

**Comment by Michael-Fletcher** [comment]
Same for me.  I deploy this using a docker CI/CD pipeline and can confirm without a shadow of a doubt that this was working with the exact same code a week ago.  It seems like RVM isn't installing all the files required.  PS. mine is a single minor version below this posts authors, so it would seem it's affecting Ruby 3.2.x builds. 

```
0.767 Fetching bundler 2.5.6
0.822 Installing bundler 2.5.6
0.906 /usr/local/rvm/rubies/ruby-3.2.3/bin//ruby_executable_hooks:15:in `read': No such file or directory @ rb_sysopen - install (Errno::ENOENT)
0.906 	from /usr/local/rvm/rubies/ruby-3.2.3/bin//ruby_executable_hooks:15:in `<main>'
```

**Comment by RailsCod3rFuture** [comment]
We found that this issue comes from the latest version of bundler gem 2.6.1. Doubt RVM can assist here. Downgrade to 2.5.9 version of bundler temporarily

**Comment by mathieujobin** [comment]
maybe 3.2.6 is fine ? 
upgrading to 3.3.6 shouldn't be a major task either...

**Comment by Michael-Fletcher** [comment]
Yeah.  My team upgraded to the latest stable of Ruby and the issue went away.  It does pose the question of what happened to the ruby installer files.  Maybe some were corrupted or missing from a CDN or something.  Maybe the issue has already fixed itself who knows.
